generic-jms-ra-jar/src/main/java/org/jboss/resource/adapter/jms/JmsConnectionFactoryImpl.java

```diff
@@ -245,9 +245,13 @@ public JMSContext createContext(String userName, String password, int sessionMod
             }
             JMSException jmse = findRootJMSException(e);
             if (jmse instanceof JMSSecurityException) {
-                throw new JMSSecurityRuntimeException(jmse.getMessage());
+                JMSSecurityRuntimeException rte = new JMSSecurityRuntimeException(jmse.getMessage());
+                rte.initCause(jmse);
+                throw rte;
             } else {
-                throw new JMSRuntimeException(e.getMessage());
+                JMSRuntimeException rte = new JMSRuntimeException(e.getMessage());
```

Comment:
same as above

Reply:
I made the change and checking JMSRuntimException it does link the initial cause so I started passing the exception in as argument 3 rather than using initCause.